### ISSUE TITLE
got pybind11 errors with Python 3.11

### ISSUE THREAD

**Issue by rth1969** [Kuree/pysv#25]
When I run the hello_word example with with Python 3.11 

I got the error with this setup: 

`Red Hat Enterprise Linux Workstation release 7.4 (Maipo), Xcelium gcc 9.3 , cmake 3.25.1 , Python 3.11.4  with  pybind11=2.11.1 and pysv=0.2.0`

the following error message:
```
-- The C compiler identification is GNU 9.3.0
-- The CXX compiler identification is GNU 9.3.0
-- Check for working C compiler: /apps/XCELIUM2303/23.03.006/tools.lnx86/cdsgcc/gcc/9.3/bin/gcc
-- Check for working C compiler: /apps/XCELIUM2303/23.03.006/tools.lnx86/cdsgcc/gcc/9.3/bin/gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /apps/XCELIUM2303/23.03.006/tools.lnx86/cdsgcc/gcc/9.3/bin/g++
-- Check for working CXX compiler: /apps/XCELIUM2303/23.03.006/tools.lnx86/cdsgcc/gcc/9.3/bin/g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- pybind11 v2.6.2 dev1
-- Found PythonInterp: <my_path>/python_venv_3.11/bin/python3.11 (found version "3.11.4") 
-- Found PythonLibs: <path to python installation>/tools/python/3.11.4/lib/libpython3.11.so
-- Performing Test HAS_FLTO
-- Performing Test HAS_FLTO - Success
-- Configuring done
-- Generating done
-- Build files have been written to: <my path>/pysv_import/build/build
Scanning dependencies of target libpysv
[ 50%] Building CXX object CMakeFiles/libpysv.dir/libpysv.cc.o
In file included from <my path>/build/pybind11/include/pybind11/attr.h:13,
                 from <my path>/build/pybind11/include/pybind11/pybind11.h:45,
                 from <my path>/build/pybind11/include/pybind11/embed.h:12,
                 from <my path>/build/libpysv.cc:1:
<my path>/build/pybind11/include/pybind11/cast.h: In function 'std::string pybind11::detail::error_string()':
<my path>/build/pybind11/include/pybind11/cast.h:446:36: error: invalid use of incomplete type 'PyFrameObject' {aka 'struct _frame'}
  446 |                 "  " + handle(frame->f_code->co_filename).cast<std::string>() +
      |                                    ^~
```

When I run the hello_world example with 3.10 it works fine. Is there a way to solve this for 3.11 ?

Thanks
Rainer

**Comment by FpgaJohn** [comment]
Search for "The [PyFrameObject](https://docs.python.org/3/c-api/frame.html#c.PyFrameObject) structure members have been removed from the public C API."

here:
https://docs.python.org/3/whatsnew/3.11.html

So you will have to use Python3.10 for now

**Comment by rth1969** [comment]
I found an ugly workaround which works for me

```
# install ofw packages
./bin/pip install pysv
./bin/pip install pdoc3

# update python versions
set FILE          =  ./lib/*/site-packages/pysv/extern/pybind11/tools/pybind11Tools.cmake
set SEARCH_STRING  =  "3.9;3.8;3.7;3.6;3.5;3.4"
set REPLACE_STRING =  "3.11;3.10;3.9;3.8;3.7;3.6;3.5;3.4"
perl -p -i.bak -e "~s|${SEARCH_STRING}|${REPLACE_STRING}|" ${FILE}

 # replace pysv/extern/pybind11/include for 3.11 Python releases
 cd ./lib/*/site-packages/pysv/extern/pybind11/
 mv include include__ORIG_PYSV_INSTALL__
 ln -s ../../../pybind11/include include
```

**Comment by Kuree** [comment]
Should be fixed with [6b01904](https://github.com/Kuree/pysv/commit/6b01904a6af86a568dcc1b8ea1deac4e901dc5b4). Please reopen this issue if not fixed.
